src/Datadog.Trace.ClrProfiler.Native/clr_helpers.cpp

```diff
@@ -299,6 +299,126 @@ std::vector<IntegrationMethod> FilterIntegrationsByCaller(
   return enabled;
 }
 
+bool IsAssemblyAvailable(
+    const AssemblyReference& wrapper_assembly,
+    const ComPtr<IMetaDataAssemblyImport>& current_assembly_import,
+    const ModuleInfo& module_info) {
+  auto found = false;
+  for (auto& assembly_ref : EnumAssemblyRefs(current_assembly_import)) {
+    const auto metadata_ref =
+        GetReferencedAssemblyMetadata(current_assembly_import, assembly_ref);
+    if (metadata_ref.name == wrapper_assembly.name &&
+        metadata_ref.version == wrapper_assembly.version) {
+      return true;
+    }
+  }
+
+  // TODO: this is the GAC in many windows scenarios
+  // We need to get the app base path from the root assembly
+  WSTRING app_directory;
+  const size_t last_slash_idx = module_info.path.rfind('\\');
+  if (std::string::npos != last_slash_idx) {
+    app_directory = module_info.path.substr(0, last_slash_idx);
+  }
+
+  const auto import_assembly_name = wrapper_assembly.name.c_str();
+
+  // We only care about finding it once
+  // TODO: though, should we worry about multiple versions?
+  const ULONG max_matches = 5;
+  const auto assembly_matches = new IUnknown*[max_matches];
+  ULONG* matching_assembly_count = 0;
+
+  auto hr = current_assembly_import->FindAssembliesByName(
```

Comment:
Currently receiving an INVALID_POINTER message, I think this is one of the last blockers aside from ensuring we pass the proper path for app_base and any private bins.

Reply:
I think it might be the `nullptr` value for `szPrivateBin`. The "private bin" folder is configurable in the `app.config` file, and is `bin` by default (at least for web apps).

See also [AppDomainSetup.PrivateBinPath](https://docs.microsoft.com/en-us/dotnet/api/system.appdomainsetup.privatebinpath).